lib/perl/Genome/Utility/IO/SeparatedValueReader.pm

```diff
@@ -89,7 +91,21 @@ sub create {
     else {
         $regexp = qr/\Q$sep\E/;
     }
-    $self->{_split} = sub{ return _strip_quotes(split($regexp, $_[0], -1)) };
+
+    if ($sep eq ',') {
+        my $csv = Text::CSV->new({ binary => 1, auto_diag => 1, sep_char => ',' });
+        $self->{_split} = sub {
+            my $line = $_[0];
+            chomp($line);
+            if ($csv->parse($line)) {
+                return _strip_quotes($csv->fields());
+            } else {
+                $self->fatal_message('Failed to parse CSV line: '. $line);
+            }
+        };
+    } else {
+        $self->{_split} = sub{ return _strip_quotes(split($regexp, $_[0], -1)) };
+    }
```

Comment:
Text::CSV can handle all kinds of cases, not just when using ```,``` as the separator.  It can also strip quotes by itself.

I'd imagine this whole module could probably get turned into a simple wrapper around Text::CSV.

Another option might be to turn this into some kind of data source factory or for UR::DataSource::FileSystem data sources or a role using UR::DS::FileSystem.  Then consumers would be able to use the same get() API as everything else, and get back real objects instead of arrays of lines.